pkgs/by-name/wo/worldpainter/package.nix

```diff
@@ -0,0 +1,71 @@
+{
+  lib,
+  stdenv,
+  fetchurl,
+  jre,
+  makeWrapper,
+  makeDesktopItem,
+  copyDesktopItems
+}:
+stdenv.mkDerivation rec {
+  pname = "worldpainter";
+  version = "2.23.2";
+
+  src = fetchurl {
+    url = "https://www.worldpainter.net/files/${pname}_${version}.tar.gz";
+    hash = "sha256-FhrfI2/mBeIZAg26Q8x8T2YU2ANwAkoC3h9Tgw6ggoY=";
+  };
+
+  nativeBuildInputs = [
+    makeWrapper
+    copyDesktopItems
+  ];
+
+  outputs = [ "out" ];
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/{bin,lib,.install4j/user}
+
+    install -Dm644 bin/*        "$out/bin/"
+    install -Dm644 lib/*        "$out/lib/"
+    install -Dm644 *.vmoptions  "$out/"
+    install -Dm755 worldpainter "$out/"
+    install -Dm755 wpscript     "$out/"
```

Comment:
Usually binaries are put in `$out/bin`

Reply:
Please also make sure to format your nix files.
You can go to the Nixpkgs root directory, run `nix-shell`, then:
nixfmt 'pkgs/by-name/wo/worldpainter/package.nix'